### PR TITLE
Update lighting effects for xbows keyboard

### PR DESCRIPTION
**Pull request by XBowsTech**
## Description

Update lighting effects,

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by XBowsTech** [comment]
@drashna This is a simple update, can you merge it for me? Thank you！

File: keyboards/xbows/knight/knight.c [hunk 1]
```diff
@@ -134,7 +134,15 @@
...
 
+ void suspend_power_down_kb(void) {
+     rgb_matrix_set_suspend_state(true);
+     suspend_power_down_user();
+ }
 
+ void suspend_wakeup_init_kb(void) {
+     rgb_matrix_set_suspend_state(false);
+     suspend_wakeup_init_user();
+ }
```
Comment (drashna) [inline keyboards/xbows/knight/knight.c:145]: These are no longer needed. 
```suggestion
```

File: keyboards/xbows/knight_plus/knight_plus.c [hunk 2]
```diff
@@ -134,7 +134,15 @@
...
 
+ void suspend_power_down_kb(void) {
+     rgb_matrix_set_suspend_state(true);
+     suspend_power_down_user();
+ }
 
+ void suspend_wakeup_init_kb(void) {
+     rgb_matrix_set_suspend_state(false);
+     suspend_wakeup_init_user();
+ }
```
Comment (drashna) [inline keyboards/xbows/knight_plus/knight_plus.c:145]:
```suggestion
```

File: keyboards/xbows/numpad/numpad.c [hunk 3]
```diff
@@ -70,6 +70,15 @@
...
 
+ void suspend_power_down_kb(void) {
+     rgb_matrix_set_suspend_state(true);
+     suspend_power_down_user();
+ }
+
+ void suspend_wakeup_init_kb(void) {
+     rgb_matrix_set_suspend_state(false);
+     suspend_wakeup_init_user();
+ }
```
Comment (drashna) [inline keyboards/xbows/numpad/numpad.c:81]:
```suggestion
```

File: keyboards/xbows/ranger/ranger.c [hunk 4]
```diff
@@ -137,6 +137,16 @@ led_config_t g_led_config = { {
...
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
```
Comment (drashna) [inline keyboards/xbows/ranger/ranger.c:149]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by XBowsTech** [comment]
Hey qmk/qmk_firmware

Thank you for contacting us. This is an automated message to let you know that we have received your message. We will get back to you as soon as possible.

Best,
The x-bows team

File: keyboards/xbows/knight/knight.c (hunk 1 above)
Comment (XBowsTech) [reply keyboards/xbows/knight/knight.c:145]: Hi, I've done some tests, and when you keep these lines of code, the keyboard works fine. My keyboard often woke up automatically when I deleted those lines.@drashna

**Comment by XBowsTech** [comment]
@drashna Can I keep these lines of code?

**Comment by XBowsTech** [comment]
Can these lines of code be preserved? I'd like to keep it if I can.@drashna @zvecr

File: keyboards/xbows/knight/knight.c (hunk 1 above)
Comment (drashna) [reply keyboards/xbows/knight/knight.c:145]: This functionality is added to the core code, which is why it's not needed. 

https://github.com/qmk/qmk_firmware/blob/474921d23187499751f14aee444cd78ae6b05a5a/tmk_core/common/avr/suspend.c#L169-L171

As for it waking automatically, I'm not sure, but that shouldn't be related to this,

**Comment by XBowsTech** [comment]
Hey qmk/qmk_firmware

Thank you for contacting us. This is an automated message to let you know that we have received your message. We will get back to you as soon as possible.

Best,
The x-bows team

**Comment by XBowsTech** [comment]
OK.I've removed redundant code. Can you merge it for me? Thank you！@drashna

**Comment by XBowsTech** [comment]
Can you merge it for me? Thank you！@fauxpark

**Review by drashna — APPROVED** [review]

**Comment by XBowsTech** [comment]
Hey qmk/qmk_firmware

Thank you for contacting us. This is an automated message to let you know that we have received your message. We will get back to you as soon as possible.

Best,
The x-bows team

**Comment by XBowsTech** [comment]
@zvecr Can you merge it for me? Thank you！

**Comment by drashna** [comment]
> @zvecr Can you merge it for me? Thank you！

https://docs.qmk.fm/#/pr_checklist?id=review-process

Also, the breaking changes cycle is almost closed (a little over a week left), so a lot of our focus is on that.

**Comment by XBowsTech** [comment]
Hey qmk/qmk_firmware

Thank you for contacting us. This is an automated message to let you know that we have received your message. We will get back to you as soon as possible.

Best,
The x-bows team

File: keyboards/xbows/knight/config.h [hunk 5]
```diff
@@ -44,6 +44,36 @@
 #    define RGB_MATRIX_CENTER \
         { 92, 33 }
 
+#    define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_LEFT_RIGHT // Sets the default mode, if none has been set
+#    define DISABLE_RGB_MATRIX_ALPHAS_MODS               // Static dual hue, speed is hue for secondary hue
```
Comment (drashna) [inline keyboards/xbows/knight/config.h:48]: Could these be switched to explicitly enabled? 

```c
#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

the code is partly there to support this, but when develop lands in a week or so, any mode not explicitly enabled will be effectively disabled.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by XBowsTech** [comment]
Hey qmk/qmk_firmware

Thank you for contacting us. This is an automated message to let you know that we have received your message. We will get back to you as soon as possible.

Best,
The x-bows team

**Comment by XBowsTech** [comment]
> > @zvecr Can you merge it for me? Thank you！
> 
> https://docs.qmk.fm/#/pr_checklist?id=review-process
> 
> Also, the breaking changes cycle is almost closed (a little over a week left), so a lot of our focus is on that.

I'm sorry to bother you with my merger, and you did a great job!

File: keyboards/xbows/knight/config.h (hunk 5 above)
Comment (XBowsTech) [reply keyboards/xbows/knight/config.h:48]: Can you merge it for me first? I will submit the update again next week.

**Pull request closed without merging**